src/Field/BareField.hpp

```diff
@@ -149,8 +159,8 @@ namespace ippl {
     }
 
     DefineReduction(Sum, sum, valL += myVal, std::plus)
-    DefineReduction(Max, max, if (myVal > valL) valL = myVal, std::greater)
-    DefineReduction(Min, min, if (myVal < valL) valL = myVal, std::less)
+    DefineReduction(Max, max, using Kokkos::max;valL = max(valL, myVal), std::greater)
+    DefineReduction(Min, min, using Kokkos::min;valL = min(valL, myVal), std::less)
```

Comment:
Considering that this is a single-line function body, the `using` alias provides no benefit. It's actually longer than if you just put `Kokkos::max`, which I would recommend.

Reply:
problem with using Kokkos::max directly is that inhibits ADL for types that don't have a max overload in the Kokkos namespace (e.g. quads, bigfloats etc. in their own namespace)

Reply:
> Considering that this is a single-line function body, the using alias provides no benefit

that's wrong, the benefit isn't brevity but possibility of ADL for overloads

Reply:
Scratch that, it is important: the reason is that for vector types, the max overload isn't in the Kokkos namespace but in ippl. Using Kokkos would break it for ippl::Vector<T, D>